File: myits-frontend/src/services/authService.ts
import api from '../config/api';
import type { LoginRequest, RegisterRequest, AuthResponse, ResetPasswordRequest, OtpSendRequest, OtpSendResponse, ApiResponse } from '../types';

export const authService = {
  login: async (credentials: LoginRequest) => {
    const response = await api.post<AuthResponse>('/auth/login', credentials);
    return response.data;
  },

  register: async (data: RegisterRequest) => {
    const response = await api.post<AuthResponse>('/auth/register', data);
    return response.data;
  },

  resetPassword: async (data: ResetPasswordRequest) => {
    const response = await api.post<ApiResponse<string>>('/auth/reset-password', data);
    return response.data;
  },

  sendOtp: async (data: OtpSendRequest) => {
    const response = await api.post<ApiResponse<OtpSendResponse>>('/auth/send-otp', data);
    return response.data;
  },

  logout: () => {
    localStorage.removeItem('token');
    localStorage.removeItem('user');
  },
};

File: myits-backend/src/main/java/com/myits/backend/service/AuthService.java
package com.myits.backend.service;

import com.myits.backend.dto.AuthResponse;
import com.myits.backend.dto.LoginRequest;
import com.myits.backend.dto.OtpSendRequest;
import com.myits.backend.dto.OtpSendResponse;
import com.myits.backend.dto.RegisterRequest;
import com.myits.backend.dto.ResetPasswordRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    void resetPassword(ResetPasswordRequest request);

    OtpSendResponse sendOtp(OtpSendRequest request);
}
